src/lib/blueprint.ts

```diff
@@ -189,19 +190,34 @@ const createEndpoint = (
   const pathParts = path.split('/')
   const endpointPath = `/${pathParts.slice(1, -1).join('/')}`
 
+  const description =
+    'description' in operation && typeof operation.description === 'string'
+      ? operation.description
+      : ''
+
+  const isUndocumented =
+    'x-undocumented' in operation && operation['x-undocumented'] !== undefined
```

Comment:
Let's normalize these checks. Effectively we are parsing the spec and using it here. The risk is that upstream `x-undocumented` is not a string as expected. Unless we validate the openapi spec before using it in this function, this function needs to protect against bad data.

We could define the `openapi.ts` schema with zod and get the valid-ion for "free", e.g.,

```
const openapiSchema = z.object(...) // have x-undocumented: z.string().optional() in the schema
const openapi = openapiSchema.parse(openapi)

// later
const isUndocumented = operation['x-undocumented']
```
 
or we need to do this manually

```
const getUndocumented = (operation) => {
  if (!('x-undocumented' in operation)) return undefined

  if (typeof operation['x-undocumented'] !== 'string') {
    console.warn('not a string!')
  } 

  return operation['x-undocumented']
}
```

Reply:
I'll work on this but I'm not sure I understand why do we need to validate the data? Isn't this supposed to be used only internally and with our public types?